### ISSUE TITLE
The function of epsilon in this algorithm

### ISSUE THREAD

**Issue by su123123123** [tusharsarkar3/XBNet#2]
I rebuild the algorithm,but at the "self.boosted_layers[i] = torch.from_numpy(np.array(self.xg.fit(x0.detach().numpy(), (self.l).detach().numpy()).feature_importances_) + self.epsilon)", I can not understand , that feature_importances_  will not change . I think that the function of epsilon does not work.

**Comment by tusharsarkar3** [comment]
Epsilon comes to the picture when the importance of a feature is 0. We take the log of feature importance in the later steps of the algorithm to maintain the same order of the weights and we want to avoid 1 as the order when we have 0 importance of the feature so we have used epsilon. Does this solve your issue ?

**Comment by su123123123** [comment]
Thank you,sir.I got it.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Tushar ***@***.***&gt;; 
发送时间: 2021年6月20日(星期天) 晚上8:28
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [tusharsarkar3/XBNet] The function of epsilon in this algorithm (#2)





 
Epsilon comes to the picture when the importance of a feature is 0. We take the log of feature importance in the later steps of the algorithm to maintain the same order of the weights and we want to avoid 1 as the order when we have 0 importance of the feature so we have used epsilon. Does this solve your issue ?
